migration/migration.go

```diff
@@ -0,0 +1,60 @@
+package migration
+
+import (
+	"context"
+	"time"
+
+	"github.com/cenk/backoff"
+	"github.com/giantswarm/microerror"
+)
+
+// Process takes a list of ordered migrators and runs them sequentially. The
+// order of migrators should never change once used to guarantee a constantly
+// stable migration path.
+func Process(ctx context.Context, migrators []Migrator) error {
```

Comment:
The idea here is to configure operators with migrators to run the migration process before the actual informer watch.